dialogue-core/src/test/java/com/palantir/dialogue/core/StickyEndpointChannels2Test.java

```diff
@@ -212,32 +295,42 @@ private TestHarness(Channel channel) {
         }
 
         TestHarness expectAddStickyTokenRequest() {
-            when(endpointChannel.execute(request)).thenAnswer((Answer<ListenableFuture<Response>>) invocation -> {
-                Request actualRequest = invocation.getArgument(0);
-                assertThat(actualRequest).isEqualTo(request);
-                LimitedChannel stickyTarget = mock(LimitedChannel.class);
-                when(stickyTarget.maybeExecute(endpoint, request, LimitEnforcement.DEFAULT_ENABLED))
-                        .thenReturn(Optional.of(responseSettableFuture));
-                return StickyAttachments.maybeAddStickyToken(
-                                stickyTarget, endpoint, actualRequest, LimitEnforcement.DEFAULT_ENABLED)
-                        .get();
-            });
+            return expectAddStickyTokenRequest(Optional.empty());
+        }
+
+        TestHarness expectAddStickyTokenRequest(Optional<Runnable> maybeAdditionalListener) {
+            when(endpointChannel.execute(Mockito.same(rawRequest)))
```

Comment:
This was quite evil too: different callbacks were getting executed, because Request has an actual #equals implementation.